.ado/templates/apple-yarn-build.yml

```diff
@@ -11,7 +11,7 @@ steps:
 
   - bash: |
       echo "Pre boot simulator"
-      UDID=$(xcrun simctl create test-iphone com.apple.CoreSimulator.SimDeviceType.iPhone-12 iOS15.5)
+      UDID=$(xcrun simctl create test-iphone com.apple.CoreSimulator.SimDeviceType.iPhone-12 iOS16.0)
```

Comment:
Should we also use iPhone-14? Not blocking.

Reply:
I can create a separate PR for this, but the important thing is getting iOS 16 support working.